src/model/file.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2022 spdx contributors
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from enum import Enum, auto
+from typing import Optional, List, Union
+
+from src.model.checksum import Checksum
+from src.model.license import License
+from src.model.license_expression import LicenseExpression
+from src.model.spdx_no_assert import SpdxNoAssert
+from src.model.spdx_none import SpdxNone
+
+
+class FileType(Enum):
+    SOURCE = auto()
+    BINARY = auto()
+    ARCHIVE = auto()
+    APPLICATION = auto()
+    AUDIO = auto()
+    IMAGE = auto()
+    TEXT = auto()
+    VIDEO = auto()
+    DOCUMENTATION = auto()
+    SPDX = auto()
+    OTHER = auto()
+
+
+class File:
+    # Mandatory properties
```

Comment:
What is mandatory and what optional depends on the SPDX version. So either we specify that we mean 2.3 here or we leave these comments out (see also document, package and snippet classes).

Reply:
removed